server/remote_cache/action_cache_server/action_cache_server.go

```diff
@@ -359,3 +370,34 @@ func (s *ActionCacheServer) maybeInlineOutputFiles(ctx context.Context, req *rep
 	}
 	return nil
 }
+
+// validateRestrictedAccess checks to see if the instance name has a restricted prefix.
+// If it does, validateRestrictedAccess uses the ClientIdentityService to assert that the
+// request comes from a trusted client: the app or an executor.
+// If the client is not trusted, the ClientIdentityService is not available, or there are any other errors,
+// validateRestrictedAccess returns an UnauthenticatedError.
+func (s *ActionCacheServer) validateRestrictedAccess(ctx context.Context, instanceName string) error {
+	if isRestricted(instanceName) {
```

Comment:
you can flip this to if !isRestricted(instanceName) { return nil } and remove one level of nesting